src/main/java/com/znsio/api/ApplitoolsInitializer.java

```diff
@@ -0,0 +1,163 @@
+import com.applitools.eyes.TestResultsStatus;
+import com.applitools.eyes.selenium.ClassicRunner;
+import com.applitools.eyes.selenium.Configuration;
+import com.applitools.eyes.visualgrid.services.RunnerOptions;
+import com.applitools.eyes.visualgrid.services.VisualGridRunner;
+import com.znsio.api.utils.Config;
+import io.appium.java_client.AppiumDriver;
+import org.apache.log4j.Logger;
+import org.openqa.selenium.WebDriver;
+import org.testng.Assert;
+import org.testng.ITestResult;
+import org.testng.annotations.*;
+import org.testng.xml.XmlTest;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+
+import static com.znsio.api.ApplitoolsConfigurationManager.*;
+
+public class ApplitoolsInitializer {
+    private static WebDriver webDriver;
+    private static AppiumDriver appiumDriver;
+    protected com.applitools.eyes.selenium.Eyes eyesOnWeb;
+    protected com.applitools.eyes.appium.Eyes eyesOnApp;
+    private static BatchInfo batch;
+    private static Configuration eyesConfig;
+    private static EyesRunner runner;
+    private static final Logger LOGGER = Logger.getLogger(ApplitoolsInitializer.class.getName());
+
+    public static void driverSetupForApplitoolsInitializer(WebDriver wDriver) {
+        webDriver = wDriver;
+    }
+
+    public static void driverSetupForApplitoolsInitializer(AppiumDriver aDriver) {
+        appiumDriver = aDriver;
+    }
+
+    @BeforeSuite
+    public void setUpApplitoolsInitializer(XmlTest suite) throws IOException, RuntimeException {
+
+        LOGGER.info("@BeforeSuite of ApplitoolsInitializer called");
+        loadProperties();
+        eyesConfig = new Configuration();
+        setTestProperties(eyesConfig);
+        setProxyIfAvailable(eyesConfig);
+        batch = new BatchInfo(eyesConfig.getAppName() + "-" + suite.getSuite().getName() + "-" + suite.getName());
+        setBatchProperties(batch);
+        eyesConfig.setBatch(batch);
+        if (isPlatformWeb() && isUltraFastGridEnabled()) {
+            setUFGBrowserConfig(eyesConfig);
+            runner = new VisualGridRunner(new RunnerOptions().testConcurrency(getConcurrency()));
+        } else {
+            runner = new ClassicRunner();
+        }
+        runner.setDontCloseBatches(true);
+    }
+
+    @BeforeMethod
+    public void initiateApplitoolsInitializer(Method method) {
+        LOGGER.info("@BeforeMethod of ApplitoolsInitializer called: " + method.getName());
+        if (isPlatformWeb()) {
+            initiateVisualWebTests(method);
+        } else {
+            initiateVisualAppTests(method);
+        }
+    }
+
+    @AfterMethod
+    public void closeApplitoolsInitializer(ITestResult iTestResult) {
+        LOGGER.info("@AfterMethod of ApplitoolsInitializer called: Waiting for visual validation results of test: " +
+                iTestResult.getName());
+        TestResults testResult;
+        if (isPlatformWeb()) {
+            testResult = eyesOnWeb.close(false);
```

Comment:
I found an issue in implementation when using eyes.closeAsync() and getting results from runner. Will discuss this over detail over a call.

Reply:
Configuring runner inside the BeforeMethod instead of BeforeSuite fixed the problem. Updated the code to use CloseAsync instead of Close and now getting the results from running instead of eyes